prog/weaveutil/attach.go

```
@@ -0,0 +1,105 @@
+package main
+
+import (
+	"fmt"
+	"net"
+	"strconv"
+	"syscall"
+
+	docker "github.com/fsouza/go-dockerclient"
+	"github.com/vishvananda/netns"
+
+	weavenet "github.com/weaveworks/weave/net"
+)
+
+func attach(args []string) error {
+	if len(args) < 4 {
+		cmdUsage("attach-container", "[--no-multicast-route] <container-id> <bridge-name> <mtu> <cidr>...")
+	}
+
+	withMulticastRoute := true
+	if args[0] == "--no-multicast-route" {
+		withMulticastRoute = false
+		args = args[1:]
+	}
+
+	pid, err := findPid(args[0])
+	if err != nil {
+		return err
+	}
+	ns, err := netns.GetFromPid(pid)
+	if err != nil {
+		return err
+	}
+	if nsToplevel, err := netns.GetFromPid(1); err != nil {
+		return fmt.Errorf("unable to open root namespace: %s", err)
+	} else if nsToplevel.Equal(ns) {
+		return fmt.Errorf("Container is running in the host network namespace, and therefore cannot be\nconnected to weave. Perhaps the container was started with --net=host.")
+	}
+	mtu, err := strconv.Atoi(args[2])
+	if err != nil && args[3] != "" {
+		return err
+	}
+	cidrs, err := parseCIDRs(args[3:])
+	if err != nil {
+		return err
+	}
+	err = weavenet.AttachContainer(ns, fmt.Sprint(pid), weavenet.VethName, args[1], mtu, withMulticastRoute, cidrs)
+	// If we detected an error but the container has died, tell the user that instead.
+	if err != nil && !processExists(pid) {
+		err = fmt.Errorf("Container %s died", args[0])
```

The original error is lost here.